### ISSUE TITLE
--commit-range ignored, needs documentation update

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/chartpress#48]
Since https://github.com/jupyterhub/chartpress/commit/55fe935e327e04d8effe522fb17711b4eb98232f, the `--commit-range` argument is no longer needed for `chartpress` to do a good job.

We should update the documentation to reflect this.

**Comment by yuvipanda** [comment]
Don't you still need commit range if the docker image gets pushed to an authenticated registry on deploy, but you want to use chartpress to build the image in PR without giving it the credentials to access the registry?

**Comment by consideRatio** [comment]
Hmm... yes. The chartpress logic currently is that a check is run if it is locally available, and if it isn't, it checks in the external docker repository. But, if you at this point don't have access to the repository for the check because you didn't want to provide such credentials, you should fall back to rebuilding it right?

This may already be what's happening, if the docker.errors.APIError is raised at least.

https://github.com/jupyterhub/chartpress/blob/9ded86b3aa9f577d90ae14fe97c8b75c8cbc2b87/chartpress.py#L141-L160

**Comment by yuvipanda** [comment]
As an example, see https://github.com/berkeley-dsep-infra/datahub/blob/235ba9f7af14e16f8b25d85799b75737c199da4b/.circleci/config.yml#L46

This never hits the docker API at all if the images haven't changed in commit-range. If we rebuilt it all the time it will slow down PRs considerably.

**Comment by consideRatio** [comment]
@yuvipanda yes I agree and care about this optimization a lot, but since the already merged commit mentioned, or chartpress 0.3.1 i believe, --commit-range is not consumed even though it is passed.

The commit-range args are parsed, and then passed to the build images function, but then never used since the commit referenced in the original post. Instead logic to set if it was locally or externally avaialable is used to determine the need for rebuild.
